### PR TITLE
feat(star-rating-select): new component

### PR DESCRIPTION
**Pull request by agliga**
## Description
* New component based on https://github.com/eBay/skin/pull/1725

## Context
* Most of the component is similar to radio. 
* __NOTE__ in draft mode because still waiting on skin part to be merged. But wanted to verify this current approach was okay.
* ~Questions: Should we add an example with fieldset? Also do we want fieldset to wrap the star-rating?~ Added fieldset example and removed `role` if there is no `a11yText`.

## References
https://github.com/eBay/ebayui-core/issues/1594

## Screenshots
Screen Shot 2022-04-20 at 3 48 48 PM

Screen Shot 2022-04-20 at 3 48 59 PM

### CODE REVIEW

File: src/components/ebay-star-rating-select/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<h1 style='display: flex; justify-content: space-between; align-items: center;'>
+    <span>
+        ebay-star-ratingselect-
```
Comment (ArtBlue) [inline src/components/ebay-star-rating-select/README.md:3]: typo `ebay-star-rating-select`

**Review by ArtBlue — APPROVED** [review]
Very nice! Great job!

File: src/components/ebay-star-rating-select/star-rating-select.stories.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    },
+
+    argTypes: {
+        disabled: {
+            control: { type: 'boolean' },
+        },
+        value: {
+            control: { type: 'number' },
+            description:
+                '1 - 5, depending on how many starts are selected. If 0 or null defaults to no stars selected',
```
Comment (ianmcburnie) [inline src/components/ebay-star-rating-select/star-rating-select.stories.js:36]: Small typo. Should be stars not starts

File: src/components/ebay-star-rating-select/examples/02-fieldset/template.marko [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class {}
+
+<fieldset>
+    <legend>Select rating</legend>
```
Comment (ianmcburnie) [inline src/components/ebay-star-rating-select/examples/02-fieldset/template.marko:4]: A small thing, but legend and aria-label strings are different. Let's consolidate on "Rate product" for both

**Comment by ianmcburnie** [comment]
I think we'll definitely need to review the size of these stars with @rbascue

**Comment by agliga** [comment]
> I think we'll definitely need to review the size of these stars with @rbascue

Yes, since in coreui we are relying only on the CSS sizes and not setting them we can probably go ahead with this PR.

**Comment by agliga** [comment]
Also verified that `aria-hidden` is set. In coreui we set it automatically when an icon does not have an `aria-label`

**Review by ianmcburnie — APPROVED** [review]

**Pull request merged**